src/iuse_actor.cpp

```diff
@@ -2978,6 +3015,31 @@ long heal_actor::finish_using( player &healer, player &patient, item &it, hp_par
         }
     }
 
+    // apply healing over time effects
+    if( bandages_power > 0 ) {
+        // remove previous effect, if exists
+        patient.remove_effect( effect_bandaged, bp_healed );
+        // add new effect
+        int bandages_intensity = get_bandaged_level( healer );
+        patient.add_effect( effect_bandaged, 1_turns, bp_healed );
+        effect &e = patient.get_effect( effect_bandaged, bp_healed );
+        e.set_duration( e.get_int_dur_factor() * bandages_intensity );
+        patient.damage_bandaged[healed] = patient.hp_max[healed] - patient.hp_cur[healed];
+        practice_amount += 2 * bandages_intensity;
+    }
+    if( disinfectant_power > 0 ) {
+        // remove previous effect, if exists
+        patient.remove_effect( effect_disinfected, bp_healed );
+        // add new effect
+        int disinfectant_intensity = get_disinfected_level( healer );
+        patient.add_effect( effect_disinfected, 1_turns, bp_healed );
+        effect &e = patient.get_effect( effect_disinfected, bp_healed );
+        e.set_duration( e.get_int_dur_factor() * disinfectant_intensity );
+        patient.damage_disinfected[healed] = patient.hp_max[healed] - patient.hp_cur[healed];
+        practice_amount += 2 * disinfectant_intensity;
+    }
```

Comment:
Copy+paste warning. Consider extracting a function to avoid two independent calculations of the same formula.

Reply:
I was not able to send effect type into new function (lack of skill), so this part was unchanged. Maybe someone will show working solution.

Reply:
You can pass it as `const efftype_id &`.

Reply:
It kind of looks like it, but it requires a lot of refactoring to make it generic, I'm sceptical it's worth it.